CMakeLists.txt

```diff
@@ -17,6 +17,10 @@ if(NOT Poco_FOUND)
     list(APPEND extra_cmake_args "-DCMAKE_CXX_FLAGS=/wd4244 /wd4530 /wd4577")
   else()
     list(APPEND extra_cmake_args "-DCMAKE_C_FLAGS=-Wno-shift-negative-value")
+    if(APPLE)
+      # Starting with macOS 10.12, the use of OSAtomic*32 funcs are deprecated.
+      list(APPEND extra_cmake_args "-DCMAKE_CXX_FLAGS=-Wno-deprecated")
```

Comment:
Was the narrower flag (`-Wno-deprecated-declarations`) you used before not working?